### PR TITLE
[Runtime][NFC] Runtime Cleanup

### PR DESCRIPTION
**Pull request by gcatron**
*Description*: Moved common types to RuntimeTypes, moved DeviceManager into runtime namespace.
*Testing*: Unit tests and resnet-runtime example pass.
*Documentation*: N/A

### CODE REVIEW

File: include/glow/Backends/DeviceManager.h [hunk 1]
```diff
@@ -27,12 +27,11 @@
 #include <string>
 
 namespace glow {
+namespace runtime {
 
 /// Callback signalling success/failure of loading a Module onto a device.
 using ReadyCBTy = std::function<void(const Module *, runtime::ResultCode)>;
```
Comment (jackm321) [inline include/glow/Backends/DeviceManager.h]: why not move this to runtime types too?

Comment (gcatron) [reply include/glow/Backends/DeviceManager.h]: We can, it's only used in the deviceManagers (and the test deviceManager in executor test) So it decided to leave it in DeviceManger.h.

**Comment by nadavrot** [comment]
@gcatron One way to communicate that this is just a refactoring change is to add the NFC label. NFC = No Functional Change. 

Example:
https://twitter.com/modocache/status/681671893963948032?lang=en

File: tests/unittests/ExecutorTest.cpp [hunk 2]
```diff
@@ -35,7 +35,7 @@ using namespace glow::runtime;
 /// registerResult() should be called before calling Executor::run() in each
 /// test. The rest of the implementation of the DeviceManager interface exists
 /// to satisfy the compiler.
-class TestDeviceManager final : public DeviceManager {
+class TestDeviceManager final : public glow::runtime::DeviceManager {
```
Comment (jackm321) [inline tests/unittests/ExecutorTest.cpp]: `glow::runtime` namespace is opened on line 30 why is this needed?

File: include/glow/Runtime/RuntimeTypes.h [hunk 3]
```diff
@@ -19,16 +19,21 @@
...
 
 namespace glow {
 namespace runtime {
 
+class DeviceManager;
 using DeviceIDTy = size_t;
 using RunIdentifierTy = size_t;
 
+/// Map of DeviceIDTy -> DeviceManager.
+using DeviceManagerMapTy = std::map<DeviceIDTy, std::shared_ptr<DeviceManager>>;
```
[the remark this answers is not in the document]
Comment (jackm321) [reply include/glow/Runtime/RuntimeTypes.h]: Perhaps a densemap would be best?

[the remark this answers is not in the document]
Comment (gcatron) [reply include/glow/Runtime/RuntimeTypes.h]: Map uses less memory for small sizes and a stable ordering is useful for the provisioner. Shared pointers are for the executor, we may want to revisit that in the future though.
